specs/ChainSpec/LightClient.md

```diff
@@ -0,0 +1,170 @@
+
+Light clients operate by periodically fetching instances of `LightClientBlockView` via particular RPC end-point desdribed [below](#rpc-end-point).
+
+Light client does't need to receive `LightClientBlockView` for all the blocks. Having the `LightClientBlockView` for block `B` is sufficient to be able to verify any statement about state or outcomes in any block in the ancestry of `B` (including `B` itself). In particular, having the `LightClientBlockView` for the head is sufficient to locally verify any statement about state or outcomes in any block on the canonical chain.
+
+However, to verify the validity of a particular `LightClientBlockView`, the light client must have verified a `LightClientBlockView` for at least one block in the preceding epoch, thus to sync to the head the light client will have to fetch and verify a `LightClientBlockView` per epoch passed.
+
+## Validating Light Client Block Views
+
+```rust
+pub enum ApprovalInner {
+    Endorsement(CryptoHash),
+    Skip(BlockHeight)
+}
+
+pub struct ValidatorStakeView {
+    pub account_id: AccountId,
+    pub public_key: PublicKey,
+    pub stake: Balance,
+}
+
+pub struct BlockHeaderInnerLiteView {
+    pub height: BlockHeight,
+    pub epoch_id: CryptoHash,
+    pub next_epoch_id: CryptoHash,
+    pub prev_state_root: CryptoHash,
+    pub outcome_root: CryptoHash,
+    pub timestamp: u64,
+    pub next_bp_hash: CryptoHash,
+}
+
+pub struct LightClientBlockView {
+    pub prev_block_hash: CryptoHash,
+    pub next_block_inner_hash: CryptoHash,
+    pub inner_lite: BlockHeaderInnerLiteView,
+    pub inner_rest_hash: CryptoHash,
+    pub next_bps: Option<Vec<ValidatorStakeView>>,
+    pub approvals_after_next: Vec<Option<Signature>>,
+}
+```
+
+Recall that the hash of the block is
+
+```rust
+sha256(concat(
+    prev_hash,
+    sha256(concat(
+        sha256(borsh(inner_lite)),
+        sha256(borsh(inner_rest))
+    ))
+))
+```
+
+The fields `prev_block_hash`, `next_block_inner_hash` and `inner_rest_hash` are used to reconstruct the hashes of the current and next block, and the approvals that will be signed, in the following way (where `block_view` is an instance of `LightClientBlockView`):
+
+```python
+def reconstruct_light_client_block_view_fields(block_view):
+    current_block_hash = sha256(concat(
+        block_view.prev_block_hash,
+        sha256(concat(
+            sha256(borsh(block_view.inner_lite)),
+            block_view.inner_rest_hash,
+        ))
+    ))
+
+    next_block_hash = sha256(concat(
+        current_block_hash,
+        block_view.next_block_inner_hash
+    ))
+
+    approval_message = concat(
+        borsh(ApprovalInner::Endorsement(next_block_hash)),
+        little_endian(block_view.inner_lite.height + 2)
+    )
+
+    return (current_block_hash, next_block_hash, approval_message)
+```
+
+The light client updates its head with the information from `LightClientBlockView` iff:
```

Comment:
```suggestion
The light client updates its head with the information from `LightClientBlockView` if:
```
Iff is a logical biconditional `<->`. If we use "iff" we mean that if points 1-6 then light client must update its head, which is not the case, it might choose to not update it.

Reply:
Then it's a light client that is not conformant to this spec.

Reply:
(similar to how when describing the block production / acceptance of full nodes, the spec will specify that if a full node receives a valid block that has higher height, it **must update**, not **may update** its head)